src/pages/search/index.js

```diff
@@ -257,27 +257,21 @@ Search.getInitialProps = async (ctx) => {
       }
     }
   } = options
-
   const filterOrders = query.filter_order ? query.filter_order.split(',') : []
-  const queryString = {
+  const queryParams = {
     ...getSearchQueryForAPI(query),
-    limit: query.size || Number(limit),
-    offset: (query.p - 1 || Number(offset)) * (query.size || Number(limit)),
+    limit: query.limit || Number(limit),
+    offset: query.offset || Number(offset) * (query.limit || Number(limit)),
     ordering: query.sortby || ordering,
     ...(query.search ? { search: query.search } : {}),
     num_downloadable_samples__gt: !query.empty
       ? Number(numDownloadableSamples.hide)
       : Number(numDownloadableSamples.show)
   }
-
-  const response = await fetchSearch(
-    queryString,
-    Number(query.p) || 1,
-    filterOrders
-  )
+  const response = await fetchSearch(queryParams, filterOrders)
 
   return {
-    query,
+    query: { ...queryParams, ...query },
```

Comment:
Wouldn't we only need to pass `queryParams` as `query` and not merge them? It should be removed when reworking the above code anyway so I won't hold up the PR for this change.

Reply:
I've removed the `limit` and `offset` from `commonQueries` in `config/options.search` and `queryParams` is now assigned as `query`. I also included a logical OR for their default values (in d54b452639ed6b33de4c8e83be24613ccd0dba12).